packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -135,10 +140,7 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, {
 			return this.workflow.versionId;
 		},
 		workflowSettings(): IWorkflowSettings {
-			if (this.workflow.settings === undefined) {
-				return {};
-			}
-			return this.workflow.settings;
+			return this.workflow.settings ?? defaults.settings;
```

Comment:
If the workflow settings are empty, we return the settings object (which would be a reference). Not sure how else it is used but I assume there is the risk that if the settings of the workflow get changed, it changes by accident also the defaults for that session.